### PR TITLE
ICU-21833 Replace NULL with nullptr in all C++ code

### PR DESCRIPTION
**Pull request by roubert**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21833
- [X] Required: The PR title must be prefixed with a JIRA Issue number.
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number.
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by roubert** [comment]
Ping?

File: icu4c/source/tools/ctestfw/unicode/ctest.h [hunk 1]
```diff
@@ -89,7 +89,7 @@ U_CDECL_END
 #define WRITE_GOLDEN_DATA_OPTION 7
 
 /**
- * Maximum amount of memory uprv_malloc should allocate before returning NULL.
+ * Maximum amount of memory uprv_malloc should allocate before returning nullptr.
```
Comment (markusicu) [inline icu4c/source/tools/ctestfw/unicode/ctest.h:92]: Optional: The ctest.h functions are C functions, so NULL is appropriate. However, this is not library code...

**Review by markusicu — COMMENTED** [review]
lgtm, please rebase & resolve merge conflicts (sorry for the long delay!)

File: icu4c/source/tools/ctestfw/unicode/ctest.h (hunk 1 above)
Comment (roubert) [reply icu4c/source/tools/ctestfw/unicode/ctest.h:92]: Done.

**Review by markusicu — APPROVED** [review]
lgtm tnx!

**Pull request merged**